### PR TITLE
Patrick/code cleanup

### PR DESCRIPTION
**Pull request by pcrumley**
This is a slightly opinionated cosmetic refactor of swift-toolbox, and some minor other fixes. Here they are.

1) Remove default features of chrono to avoid https://rustsec.org/advisories/RUSTSEC-2020-0071.html
2) Restructure modules to be in `name/mod.rs`. To me, this style is cleaner, I am happy to revert this change.
3) Make all doc strings of struct and enums follow rustdoc syntax, so you should be able to see the fields in your favorite IDE & they are properly formatted in the rustdocs. (FYI the way I found them was just run `cargo doc --document-private-items` and just look over the docs that were created. It was useful to help me get acclimated to this crate anyway. Closes https://swift-nav.atlassian.net/browse/CPP-348
4) Fix all markdown warning thrown by rustdoc.
5) Update `utils/mod.rs` unit tests to use a case based test framework, have the message thrown if test fails be more helpful when comparing floats & byte strings.

Sorry to put these all together, I should have done them as separate PRs.

### CODE REVIEW

File: console_backend/src/shared_state.rs [hunk 1]
```diff
@@ -680,9 +680,6 @@ impl SettingsTabState {
 // Navbar Types.
 
 /// Directory struct for storing informating and helpers pertaining to project directory.
-///
-/// Taken from swift-cli/swift/src/types.rs.
-/// impl taken from swift-cli/swift/src/lib.rs.
```
Comment (pcrumley) [inline console_backend/src/shared_state.rs:685]: remove comment that references internal code, and it may have drifted by now anyway

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/3859117730)

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/3898367291)

**Review by adrian-kong — APPROVED** [review]
🚀

File: console_backend/src/tabs/solution_tab/solution_position_tab.rs [hunk 2]
```diff
@@ -424,7 +419,7 @@ impl SolutionPositionTab {
 
     /// Handle PosLLH / PosLLHDepA messages.
     ///
-    /// TODO(johnmichael.burke@) https://swift-nav.atlassian.net/browse/CPP-95
+    /// TODO(johnmichael.burke@) <https://swift-nav.atlassian.net/browse/CPP-95>
```
Comment (pcrumley) [inline console_backend/src/tabs/solution_tab/solution_position_tab.rs:422]: the TODO is also marked done. deleting it as well

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/3906862082)

File: console_backend/src/tabs/solution_tab/solution_position_tab.rs (hunk 2 above)
Comment (pcrumley) [reply console_backend/src/tabs/solution_tab/solution_position_tab.rs:422]: i got distracted by some rustdoc linting issues, but i turns out this isn't quite done even though it is marked as done in JIRA... leaving it for now.

**Pull request merged**